content/geometry/HullDiameter.h

```diff
@@ -0,0 +1,22 @@
+/**
+ * Author: Oleksandr Bacherikov, chilli
+ * Date: 2019-05-05
+ * License: CC0
+ * Source: https://codeforces.com/blog/entry/48868
+ * Description: Returns the two points with max distance on a convex hull. If input isn't a convex hull, call "convexHull" first.
+ * Works with integers.
+ * Status: Fuzz-tested, submitted on kattis roberthood
+ */
+#pragma once
+
+array<P, 2> hullDiameter(vector<P> S) {
+	int n = sz(S), j = n < 2 ? 0 : 1;
+	pair<ll, array<P, 2>> res;
```

Comment:
this needs to be initialized to something

Reply:
https://stackoverflow.com/questions/9025792/does-the-default-constructor-of-stdpair-set-basic-types-int-etc-to-zero

Reply:
oh, interesting. might still want to return `{S[0], S[0]}` if `sz(S) == 1` though... or assert `sz(S) >= 2`